### PR TITLE
Careers templates should extend base-protocol.html (Fixes #11027)

### LINKED ISSUE #11027: Careers templates should extend base-protocol.html
## Description

Right now the careers pages are using their own unique base template, which means we're missing out on a lot of the standard things that other bedrock pages inherit for free, such as analytics setup, sentry error reporting, standardized metadata etc.

Even though the pages are using an outdated version of Protocol, looking at the template I don't see any reason why they can't be using our standard base template. It should not impact the current design at all.

---

#### Success Criteria
- [x] Careers pages are updated to inherit from the standard bedrock base template.

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
- Updates careers pages to extend from the standard `base-protocol-mozilla.html` template.
- Only exclude minified JS bundles from linting/prettier.

## Issue / Bugzilla link
#11027

## Testing
Demo: https://www-demo1.allizom.org/en-US/careers/

- [ ] Pages still look / function the same.
- [ ] GA is still loaded
- [ ] Sentry JS is now also loaded

### CODE REVIEW

File: bedrock/careers/templates/careers/base.html [hunk 1]
```diff
@@ -4,74 +4,50 @@
...
+{% block site_footer %}
+  {% include "careers/includes/footer.html" %}
+{% endblock %}
+
+{% block js %}
+  <script src="{{ static('js/careers/libs/protocol-navigation.min.js') }}"></script>
+  <script src="{{ static('js/careers/libs/protocol-menu.min.js') }}"></script>
+  {{ js_bundle('careers-base') }}
+{% endblock %}
+>>>>>>> 4b0f4a7d2 (Careers templates should extend base-protocol.html (Fixes #11027))
```
Comment (stevejalim) [inline bedrock/careers/templates/careers/base.html:53]: @alexgibson Spotted this while exoploring some PRs. Simple deletion enough to remedy it?
```suggestion
```

Comment (alexgibson) [reply bedrock/careers/templates/careers/base.html:53]: Bah, thanks

File: .prettierignore [hunk 2]
```diff
@@ -15,7 +15,7 @@ media/js/libs/**/
 media/js/ie/libs/**/
 
 # Ignore careers JS files for now
```
Comment (stevejalim) [inline .prettierignore:17]:
```suggestion
# Ignore pre-built careers JS files for now
```

File: bedrock/careers/templates/careers/position.html [hunk 3]
```diff
@@ -6,21 +6,21 @@
 
 {% extends "careers/base.html" %}
 
-{% set meta_title = "Mozilla Careers &mdash; " + position.title + " &mdash; Open Positions" %}
-{% set meta_description = "Mozilla Careers &mdash; " + meta_description|truncate(175, end='…') %}
+{% block page_title %}Mozilla Careers — {{ position.title }} — Open Positions{% endblock %}
+{% block page_desc %}meta_description|truncate(175, end='…'){% endblock %}
```
Comment (stevejalim) [inline bedrock/careers/templates/careers/position.html:10]: This doesn't appear to be working, rendering the following instead

`<meta name="description" content="meta_description|truncate(175, end=&#39;…&#39;)">`
eg view-source:https://www-demo1.allizom.org/en-US/careers/position/gh/3596930/

File: media/js/careers/filters.js [hunk 4]
```diff
@@ -30,17 +43,18 @@
         /**
          * When a filter changes, refresh the position list.
          */
-        onFilterChange: function() {
+        onFilterChange: function () {
             var i;
             // collection of tr.position elements
-            var positions = this.positionTable.getElementsByClassName('position');
+            var positions =
+                this.positionTable.getElementsByClassName('position');
```
Comment (stevejalim) [inline media/js/careers/filters.js:50]: Odd for `prettier` to break a line like this? Or are we running a shorter line-length than the 150chars we use for Python? (Fine if so, just wondering...)

File: media/static-bundles.json [hunk 5]
```diff
@@ -1650,9 +1650,16 @@
     },
     {
       "files": [
-        "js/base/dnt-helper.js"
+        "js/careers/base.js"
       ],
-      "name": "careers-dnt-helper"
```
Comment (stevejalim) [inline media/static-bundles.json:1655]: Has the DNT helper gone away entirely?

**Review by stevejalim — CHANGES_REQUESTED** [review]
Works pretty well 🎸 , but I did spot a bug with the `desc` attr in the head that's worth a fix.

Separately, I've noticed that filtering by `Remote` shows no results (here _and_ in production) even though `Remote` appears to be a category. We could fix that here, or split it out to more of a hotfix - depends how you feel about this branch.

File: media/js/careers/filters.js (hunk 4 above)
Comment (alexgibson) [reply media/js/careers/filters.js:50]: Line length in prettier is a rough limit of ~80 characters for readability. It is one of the few configurable options, but I'm not going down that path here 😉 

https://prettier.io/docs/en/options.html#print-width

File: media/static-bundles.json (hunk 5 above)
Comment (alexgibson) [reply media/static-bundles.json:1655]: It's already included in the base template bundle, so no need to bundle it again separately any more :)

File: bedrock/careers/templates/careers/position.html (hunk 3 above)
Comment (alexgibson) [reply bedrock/careers/templates/careers/position.html:10]: Good catch, will fix

**Comment by alexgibson** [comment]
> Separately, I've noticed that filtering by Remote shows no results (here and in production) even though Remote appears to be a category. We could fix that here, or split it out to more of a hotfix - depends how you feel about this branch. See https://www.mozilla.org/en-US/careers/listings/?location=Remote for an example

This seems unrelated to this issue / PR, so best to deal with it in a separate issue I think 👍

**Comment by alexgibson** [comment]
@stevejalim thanks for the review, updated

**Review by stevejalim — APPROVED** [review]
Nice one! r+ 🎅

**Comment by stevejalim** [comment]
@alexgibson Hold off merging this - there may be a problem loading the JS

**Comment by stevejalim** [comment]
Checked locally again just now and no 404s after a clean rebuild --> 🚢

**Pull request merged**